### PR TITLE
Implement SQL LENGTH function

### PR DESCRIPTION
**Pull request by bennostein**
This is another thing I found myself wishing for yesterday, pretty straightforward to add.

### CODE REVIEW

File: sql/test/unit/test_length.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+create table t (x text, y text, z text);
+
+insert into t values ('', null, 'test');
+
+select x, length(x), y, length(y), z, length(z) from t;
```
Comment (gregsexton) [inline sql/test/unit/test_length.sql]: could we also test a literal? `length('foo')`

**Review by gregsexton — COMMENTED** [review]
Awesome, thanks for adding this!

File: sql/src/SqlCAst.sk [hunk 2]
```diff
@@ -451,6 +451,7 @@ class CAnd(CExpr<Int>, CExpr<Int>) extends CExpr<Int>
 class CUp<T: frozen>(CExpr<T>) extends CExpr<T>
 class CCoalesce<T: frozen>(Array<CExpr<T>>) extends CExpr<T>
 class CIIsNull(CExpr<Int>) extends CExpr<Int>
+class CILength(CExpr<String>) extends CExpr<Int>
```
Comment (mbouaziz) [inline sql/src/SqlCAst.sk:454]: Should it be named `CSLength`?

Comment (bennostein) [reply sql/src/SqlCAst.sk:454]: Ahh, good catch, I misinterpreted the convention here as that letter denoting the return type rather than the arg type.

**Pull request merged**